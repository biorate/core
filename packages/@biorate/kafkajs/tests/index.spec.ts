import { expect } from 'chai';
import { range } from 'lodash';
import { root, topic } from './__mocks__';

describe('@biorate/kafkajs', function () {
  this.timeout(Infinity);

  before(async () => {
    await root.$run();
  });

  it('Admin #createTopics', async () => {
    await root.admin!.current!.createTopics({
      topics: [{ topic, numPartitions: 1 }],
    });
    expect(
      await root.admin!.current!.fetchTopicMetadata({ topics: [topic] }),
    ).toMatchSnapshot();
  });

  it('Admin #createPartitions', async () => {
    await root.admin!.current!.createPartitions({
      topicPartitions: [
        {
          topic,
          count: 3,
        },
      ],
    });
    expect(
      await root.admin!.current!.fetchTopicMetadata({ topics: [topic] }),
    ).toMatchSnapshot();
  });

  it('Producer #send', async () => {
    await root.producer!.send('producer', {
      topic,
      messages: range(0, 20000).map((i) => ({
        key: `key ${i}`,
        value: `hello world ${i}`,
      })),
    });
  });

  // it('Producer #transaction', async () => {
  //   const connection = root.producer!.get('producer');
  //   const transaction = await connection.transaction();
  //   console.log(1);
  //   await transaction.send({
  //     topic,
  //     messages: range(0, 10).map((i) => ({
  //       key: `key ${i}`,
  //       value: `hello world ${i}`,
  //     })),
  //   });
  //   console.log(2);
  //   await transaction.commit();
  //   console.log(3);
  // });

  it('Consumer #run', (done) => {
    let count = 0;
    root.consumer!.subscribe('consumer', async (messages) => {
      count += messages.length;
      if (count === 20000) done();
    });
  });

  it('Admin #deleteTopics', async () => {
    await root.admin!.current!.deleteTopics({ topics: [topic] });
  });
});
