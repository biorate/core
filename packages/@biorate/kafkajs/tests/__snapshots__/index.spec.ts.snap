// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`@biorate/kafkajs Admin #createPartitions 1`] = `
Object {
  "topics": Array [
    Object {
      "name": "kafkajs",
      "partitions": Array [
        Object {
          "isr": Array [
            1,
          ],
          "leader": 1,
          "offlineReplicas": Array [],
          "partitionErrorCode": 0,
          "partitionId": 0,
          "replicas": Array [
            1,
          ],
        },
        Object {
          "isr": Array [
            1,
          ],
          "leader": 1,
          "offlineReplicas": Array [],
          "partitionErrorCode": 0,
          "partitionId": 2,
          "replicas": Array [
            1,
          ],
        },
        Object {
          "isr": Array [
            1,
          ],
          "leader": 1,
          "offlineReplicas": Array [],
          "partitionErrorCode": 0,
          "partitionId": 1,
          "replicas": Array [
            1,
          ],
        },
      ],
    },
  ],
}
`;

exports[`@biorate/kafkajs Admin #createTopics 1`] = `
Object {
  "topics": Array [
    Object {
      "name": "kafkajs",
      "partitions": Array [
        Object {
          "isr": Array [
            1,
          ],
          "leader": 1,
          "offlineReplicas": Array [],
          "partitionErrorCode": 0,
          "partitionId": 0,
          "replicas": Array [
            1,
          ],
        },
      ],
    },
  ],
}
`;
